### PR TITLE
Devops 1719 multi slack channels

### PR DESCRIPTION
**Pull request by rc-marina**

### CODE REVIEW

File: pkg/controller/summon/components/notification.go [hunk 1]
```diff
@@ -185,6 +185,17 @@ func (c *notificationComponent) handleSuccess(instance *summonv1beta1.SummonPlat
 		}
 	}
 
+	// Send to additional slack channels.
+	if len(instance.Spec.Notifications.SlackChannels) != 0 {
```
Comment (coderanger) [inline pkg/controller/summon/components/notification.go]: Do you need to check the len first? Usually range on an empty slice should be safe I think.

File: pkg/controller/summon/controller_notification_test.go [hunk 2]
```diff
@@ -69,6 +74,21 @@ var _ = Describe("Summon controller", func() {
 				},
 			},
 		}
+
+		// Set up Slack client with the test user credentials and find the most recent message.
+		slackClient = slack.New(os.Getenv("SLACK_API_KEY"))
```
Comment (coderanger) [inline pkg/controller/summon/controller_notification_test.go]: You took out the Skip stuff, which is still needed or anyone running these tests on their workstation without the API key will get failures.

File: pkg/controller/summon/components/notification.go (hunk 1 above)
Comment (rc-marina) [reply pkg/controller/summon/components/notification.go]: you were right -- we dont need the if check

File: pkg/controller/summon/controller_notification_test.go (hunk 2 above)
Comment (rc-marina) [reply pkg/controller/summon/controller_notification_test.go]: Added skip back in with following commit c05363f

**Comment by rc-marina** [comment]
After discussing with @coderanger about the tradeoffs of saving time on running Slack and Deployment-status tests (by combining their Contexts) vs having cleaner code if we were to separate them using Contexts, it didn't seem worthwhile to complicate the code unless it shaved off more than a min of test run time. 

For comparison:
If Slack and Deployment-status test are split into context, my local test times (focused on controller_notification_test) are:

```
Ran 8 of 17 Specs in 84.076 seconds
SUCCESS! -- 8 Passed | 0 Failed | 0 Pending | 9 Skipped
PASS | FOCUSED

Ginkgo ran 1 suite in 1m31.169810504s
Test Suite Passed
```
If combined:
```Ran 4 of 13 Specs in 48.055 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 9 Skipped
PASS | FOCUSED

Ginkgo ran 1 suite in 53.594165424s
```

**Pull request merged**